physical/physical_test.go

```diff
@@ -189,6 +189,21 @@ func testBackend_ListPrefix(t *testing.T, b Backend) {
 	if keys[0] != "baz" {
 		t.Fatalf("bad: %v", keys)
 	}
+
+	// Delete should recursively remove paths
+	err = b.Delete("foo")
```

Comment:
This should not be the case actually. This is invalid. If "foo" is deleted, only that key should be. The directories under should not be affected.